File: edexOsgi/build.edex/opt/esb/js/GridRequest.js
function GridRequest(){
  this.plugin = "grib";
  this.subscribe = false;
  this.subscription = null;
  this.queryResults = null;
  this.createImage = false;
  this.reproject = false;
  this.colormap = "BW";
  this.format = "png";
  this.scaleFactor = 1.0;
  this.longitude = -91.0;
  this.latitude = 42.0;
  this.slice = false;
  this.query = new TermQuery(this.plugin, subscriptionDataFieldId, subscriptionDataQueryId);
}

function _addParameter(name, value, operand){
  	if(arguments.length==2){
		this.query.addParameter(name,value);
	} else{
		this.query.addParameter(name,value,operand);
	}
}

function _addList(name, value){
  this.query.addList(name, value);
}

function _setCount(count){
  this.query.setCount(count);
}

function _setScaleFactor(scale){
  this.scaleFactor = scale;
}

function _enableSubscription(){
  this.subscribe = true;
}

function _reprojectImage(reproject){
  this.reproject = reproject;
}

function _setColormap(colormap){
  this.colormap = colormap;
}

function _setFormat(format){
  this.format = format;
}

function _setSortValue(sortValue){
  this.query.setSortBy(sortValue);
}

function _requestImage(image){
  this.createImage = image;
}
function _requestSlice(slice){
  this.createSlice = slice;
}

function _setLatitude(latitude){
  	this.latitude = latitude;
}
function _setLongitude(longitude){
	this.longitude = longitude;
}
function _execute()
{
  if(this.subscribe){
    this.subscription = new Subscription();
    this.subscription.setup(this.query);
  }
  this.queryResults = this.query.execute();
  if(this.queryResults == null || this.queryResults.size() == 0)
  {
    var response = new MakeResponseNull("Query returned 0 results.",this.query);
    return response.execute();
  }
  else
  {
  
  	if(this.createSlice) {
      return this.makegridslicerResponse();
    }

    if(this.createImage){
      return this.makeImageResponse();
    }
    else{
      return this.makeXmlResponse();
    }  }
}

function _makeXmlResponse()
{
  var count = this.queryResults.size();
  var response = new Array(count);
  for(i=0; i < count; i++)
  {
    var currentQuery = this.queryResults.get(i);
    var fileIn = new FileIn(this.plugin, currentQuery);
    var extractor = new GribExtractData(fileIn.execute(),currentQuery);
    extractor.execute();
    
    var makeResponse = new MakeResponseXml(currentQuery);
    response[i] = makeResponse.execute();
  }
  return response;
}

function _makeImageResponse(){
  var response = new Array();
  for(i=0; i < this.queryResults.size(); i++)
  {  
    var currentQuery = this.queryResults.get(i);
    var geom = currentQuery.getSpatialInfo().getMapGeom();
    var crs = currentQuery.getSpatialInfo().getCrsObject();
    var fileIn = new FileIn(this.plugin, currentQuery);
    var extractor = new GribExtractData(fileIn.execute(),currentQuery);    
    var gribMap = new GribMap(this.plugin, this.colormap, extractor.execute(), geom);
    gribMap.setScaleFactor(this.scaleFactor);
    var imageData = gribMap.execute();
    geom = gribMap.getGridGeometry();
    var colorMap = new ColorMapImage(this.colormap, imageData, geom);
    var imageOut = null;
    if(this.reproject){
      var reproject = new ReprojectImage(colorMap.execute(), geom, crs);
      var reprojectedImage = reproject.execute();
      imageOut = new ImageOut(reprojectedImage, this.format, reproject.getGridGeometry());
    }
    else
    {
      imageOut = new ImageOut(colorMap.execute(), this.format,geom);
    }
    var fileOut = new FileOut(imageOut.execute(), this.format);
    var writeFile = fileOut.execute();
    var makeResponse = new MakeResponseUri(writeFile, null, currentQuery.getIdentifier(), this.format);
    response[i] = makeResponse.execute();
  }
  return response;
}

GridRequest.prototype.execute = _execute;
GridRequest.prototype.makeXmlResponse = _makeXmlResponse;
GridRequest.prototype.makeImageResponse = _makeImageResponse;
GridRequest.prototype.addParameter = _addParameter;
GridRequest.prototype.addList = _addList;
GridRequest.prototype.setCount = _setCount;
GridRequest.prototype.setScaleFactor = _setScaleFactor;
GridRequest.prototype.enableSubscription = _enableSubscription;
GridRequest.prototype.reprojectImage = _reprojectImage;
GridRequest.prototype.setColormap = _setColormap;
GridRequest.prototype.setFormat = _setFormat;
GridRequest.prototype.setSortValue = _setSortValue;
GridRequest.prototype.requestImage = _requestImage;
GridRequest.prototype.setLatitude = _setLatitude;
GridRequest.prototype.setLongitude = _setLongitude;

File: edexOsgi/build.edex/esb/data/utility/edex_static/base/python/GridRequest.py
##
# This software was developed and / or modified by Raytheon Company,
# pursuant to Contract DG133W-05-CQ-1067 with the US Government.
# 
# U.S. EXPORT CONTROLLED TECHNICAL DATA
# This software product contains export-restricted data whose
# export/transfer/disclosure is restricted by U.S. law. Dissemination
# to non-U.S. persons whether in the United States or abroad requires
# an export license or other authorization.
# 
# Contractor Name:        Raytheon Company
# Contractor Address:     6825 Pine Street, Suite 340
#                         Mail Stop B8
#                         Omaha, NE 68106
#                         402.291.0100
# 
# See the AWIPS II Master Rights File ("Master Rights File.pdf") for
# further licensing information.
##

import BaseRequest
from java.util import ArrayList
from com.raytheon.edex.uengine.tasks.decode import FileIn

#
# Request of grid image script  
#
#    
#     SOFTWARE HISTORY
#    
#    Date            Ticket#       Engineer       Description
#    ------------    ----------    -----------    --------------------------
#    04/14/08                      njensen       Initial Creation.
#    
#
 
class GridRequest(BaseRequest.BaseRequest):
    
    def __init__(self, pluginName):
        BaseRequest.BaseRequest.__init__(self, pluginName)
        self.__createImage = False
        self.__reproject = False
        self.__colormap = "BW"
        self.__format = "png"
        self.__scaleFactor = 1
        self.__kml = False

    def setColormap(self, colormap):
        self.__colormap = colormap
        
    def setFormat(self, format):
        self.__format = format
    
    def setScaleFactor(self, scale):
        self.__scaleFactor = scale
    
    def reprojectImage(self, reproject):
        self.__reproject = reproject
    
    def requestImage(self, image):
        self.__createImage = image
        
    def requestKml(self, kml):
        self.__kml = kml

    def execute(self):        
        self.queryResults = self.query.execute()
        if self.queryResults is None or self.queryResults.size() == 0:
            return self.makeNullResponse()
        else:
            if self.__createImage:
                return self.__makeImageResponse()
            else:
                return self.makeResponse()
    
    def makeResponse(self):        
        from com.raytheon.uf.common.message.response import ResponseMessageGeneric
        count = self.queryResults.size()
        response = ArrayList()
        for i in range(count):
            currentQuery = self.queryResults.get(i)          
            response.add(ResponseMessageGeneric(currentQuery))
        return response
    
    def makeNullResponse(self):        
        response = ArrayList()
        return response
    
    def __makeImageResponse(self):                
        from com.raytheon.edex.uengine.tasks.grib import GribMap
        from com.raytheon.edex.uengine.tasks.process import ColorMapImage, ReprojectImage, ImageOut
        from com.raytheon.edex.uengine.tasks.output import FileOut
        from com.raytheon.edex.uengine.tasks.response import MakeResponseUri
        from com.raytheon.uf.common.geospatial import MapUtil
        response = ArrayList()
        size = self.queryResults.size()
        for i in range(size):
            currentQuery = self.queryResults.get(i)
            geom = MapUtil.getGridGeometry(currentQuery.getSpatialObject())
            crs = geom.getCoordinateReferenceSystem()
            fileIn = FileIn(self.plugin, currentQuery)
            gribMap = GribMap(self.plugin, self.__colormap, fileIn.execute(), geom)
            gribMap.setScaleFactor(self.__scaleFactor)
            imageData = gribMap.execute()
            geom = gribMap.getGridGeometry()
            colorMap = ColorMapImage(self.__colormap, imageData, geom)
            imageOut = None
            if self.__reproject:
                reproject = ReprojectImage(colorMap.execute(), geom, crs)
                reprojectedImage = reproject.execute()
                imageOut = ImageOut(reprojectedImage, self.__format, reproject.getGridGeometry())
            else:
                imageOut = ImageOut(colorMap.execute(), self.__format,geom)
            fileOut = FileOut(imageOut.execute(), self.__format)
            writeFile = fileOut.execute()
            makeResponse = MakeResponseUri(writeFile, None, currentQuery.getIdentifier(), self.__format)
            response.add(makeResponse.execute())
            if self.__kml:
                from com.raytheon.edex.uengine.tasks.output import KmlImage
                kmlImage = KmlImage(writeFile, geom)
                kmlFile = kmlImage.execute()
                kmlResponse = MakeResponseUri(kmlFile, None, None, "kml")
                response.add(kmlResponse.execute())

        return response
                